### PR TITLE
[#22] Use o'clock, remove time funcs

### LINKED ISSUE #22: Move 'threadDelay' and 'time-units' utils into separate package
Often in projects you want something small from `serokell-util`. But you don't want to add `serokell-util` into dependencies because it's a huge bardaq and very big dependency. It would be nice to split `serokell-util` into smaller pieces. So let's begin this process.

I think 'time-units' utils and especially this function:

```haskell
-- | Convenient version of Control.Concurrent.threadDelay which takes
-- any time-unit and operates in any MonadIO
threadDelay :: (MonadIO m, TimeUnit unit) => unit -> m ()
threadDelay = liftIO . Concurrent.threadDelay . fromIntegral . toMicroseconds
```

should be moved into separate package. It can be our fork of 'time-units' or we can create new 'time-units' redesigning existing one (using new GHC 8 features). This 'threadDelay' function is very useful. And working with `Integer` as a time is very sad(((

P.S. This PR is motivated by reviewing code which works with `Integer` time. Such code is not pleasant to write, easy to mess up, very hard to review.

### PR DESCRIPTION
**Pull request by vrom911**
Resolves #22 .

### CODE REVIEW

File: src/Serokell/Util/Bench.hs [hunk 1]
```diff
@@ -9,26 +18,22 @@ module Serokell.Util.Bench
...
-                                      getTime, toNanoSecs)
+import Universum
+
+import Fmt (Buildable (build), (+||), (||+))
+import System.Clock (Clock (..), TimeSpec, diffTimeSpec, getTime, toNanoSecs)
+import Time (Time, ns, toUnit, KnownRat)
 
 -- | Get current wall-clock time as any time unit.
-getWallTime :: (MonadIO m, TimeUnit a) => m a
+getWallTime :: (MonadIO m) => forall unit . KnownRat unit => m (Time unit)
```
Comment (ChShersh) [inline src/Serokell/Util/Bench.hs]: Type signature of this function is a little bit weird... You don't need to use `Rank2Types` extension, you need to specify all type variables before first constraint, like `forall unit m . ...`.

Comment (vrom911) [reply src/Serokell/Util/Bench.hs]: :scream:

**Comment by ChShersh** [comment]
@gromakovsky Could you please review this when you have time?

File: src/Serokell/Util/Bench.hs [hunk 2]
```diff
@@ -1,6 +1,14 @@
+{-# LANGUAGE CPP                 #-}
+{-# LANGUAGE ExplicitForAll      #-}
+{-# LANGUAGE TypeApplications    #-}
```
Comment (gromakovsky) [inline src/Serokell/Util/Bench.hs]: AFAIU, `TypeApplications` is enabled by default.

**Pull request merged**